lib/floe/workflow/runner/docker.rb

```diff
@@ -30,12 +30,11 @@ def run_async!(resource, env = {}, secrets = {})
 
           begin
             runner_context["container_ref"] = run_container(image, env, runner_context["secrets_ref"])
-          rescue
+            runner_context
+          rescue => err
             cleanup(runner_context)
-            raise
+            {"Error" => "States.TaskFailed", "Cause" => err.to_s}
```

Comment:
Would it help with clarity to also include the error class?

```suggestion
            {"Error" => "States.TaskFailed", "Cause" => "#{err.class.name}: #{err}"}
```

Reply:
Sure, done

Reply:
Hm thinking about this a little more, not sure how helpful the ruby exception class is going to be to the user since this is floe internal stuff.

What if we are more explicit about catching e.g. `Kubeclient::Error` and `AwesomeSpawn::CommandResultError` and just print the error string but let other exceptions raise up.

Reply:
great suggestion

Reply:
> What if we are more explicit about catching e.g. Kubeclient::Error and AwesomeSpawn::CommandResultError and just print the error string but let other exceptions raise up.

Not sure - the ultimate problem is that the parent wasn't handling it at the task level at it just got stuck.  Perhaps we need changes for handling other errors on that side as well?

Reply:
Well the problem that we saw on kubernetes was that the pod status was Pending so even though it "failed" we treated it as if it hadn't started yet (since that status was also Pending).

If an unhandled exception is raised the MiqQueue deliver method will catch it and still invoke the `queue_callback` to mark the task as failed (I just tested this by throwing a raise in the workflow run_nonblock).

I'd like for user errors like podman/k8s failed to pull an image be handled as ASL errors, and anything else like NoMethodError on NilClass (straight bugs) be raised up as exceptions so we get a backtrace in the logs.

Reply:
ok yeah I agree :+1: